include/boost/geometry/strategies/spherical/distance_cross_track_box_box.hpp

```diff
@@ -45,11 +47,11 @@ class cross_track_box_box_generic
 {
 public :
 
-    template <typename Point, typename PPStrategy, typename PSStrategy>
-    ReturnType static inline diagonal_case(Point topA,
-                                           Point topB,
-                                           Point bottomA,
-                                           Point bottomB,
+    template <typename Point1, typename Point2, typename PPStrategy, typename PSStrategy>
+    ReturnType static inline diagonal_case(Point1 topA,
+                                           Point2 topB,
+                                           Point1 bottomA,
+                                           Point2 bottomB,
```

Comment:
Thanks, good to see a PR from your side!

So the point types can be different, I noticed. That's fine.
But they can be made `Point1 const&` (it was already before this PR - but now it's touched...), and the strategies too.
Also unusual is that `ReturnType` is specified before `static inline` but that was also already the case.

Anyway, I'me fine with the changes.

Reply:
Done.